### PR TITLE
feat: add css color name support

### LINKED ISSUE #10: Add support for CSS color names
I just saw that you already added Xterm colors. Would be cool to also have CSS color name support. Here is a complete list of all colors: https://www.w3schools.com/cssref/css_colors.asp

If you would help me out on what I'd have to add where, I'd me happy to open a PR :)

### PR DESCRIPTION
**Pull request by SirWindfield**
Closes #10

### CODE REVIEW

File: src/dyn_colors.rs [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #[allow(unused_imports)]
-use crate::{AnsiColors, BgDynColorDisplay, DynColor, FgDynColorDisplay, Rgb, XtermColors};
+use crate::{
+    colors::css::CssColors, AnsiColors, BgDynColorDisplay, DynColor, FgDynColorDisplay, Rgb,
```
Comment (SirWindfield) [inline src/dyn_colors.rs:3]: Not entirely sure how to get the color names into the root namespace of the library. I mean I know how but I can't find the piece of code inside the crate that does it for the other color implementations (like xterm).

File: src/tests.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 use super::colors::*;
 use super::OwoColorize;
+use crate::colors::css::Lavender;
```
Comment (SirWindfield) [inline src/tests.rs:3]: > Not entirely sure how to get the color names into the root namespace of the library. I mean I know how but I can't find the piece of code inside the crate that does it for the other color implementations (like xterm).

Same spiel

**Comment by SirWindfield** [comment]
Would you be open to an internal refactoring that puts each color implementation into its own file? Currently ANSI lives inside the module's scope instead of its own. It took me some while to understand what where is.

On a side note: `Color` doesn't have to be implemented for the CssColors right? As they use a type definition relying on `CustomColor` and that one is already implementing `Color` (I think?).

**Comment by jam1garner** [comment]
> Would you be open to an internal refactoring that puts each color implementation into its own file?

I remember there being some issue towards this, but if it wouldn't be too much effort I would accept this.

> On a side note: Color doesn't have to be implemented for the CssColors right? As they use a type definition relying on CustomColor and that one is already implementing Color (I think?).

Correct

> Would you be interested in a new PR with some basic GitHub Actions workflows?

Clippy, fmt, and tests sounds great :+1:

**Comment by jam1garner** [comment]
Added a commit to your PR to fix your issue regarding how to export from the top level of the crate. The trick is to stick the type in a `pub(crate)` module before re-exporting.

As for "having colors in the root of the library" -> xterm and css colors don't go into the root of the library? They go under their respective module, which is the intention. Having the colors on the root would likely be too much noise when looking at the docs. This also allows you to glob import all of a specific color if you so choose (esp. useful for re-exporting all CSS colors, for example)

File: src/colors.rs [hunk 3]
```diff
@@ -170,6 +170,7 @@ impl_fmt_for_dyn! {
     (BgDynColorDisplay, fmt::Pointer,  fmt_ansi_bg),
 }
 
+pub mod css;
```
Comment (jam1garner) [inline src/colors.rs:173]: Document this module

**Review by jam1garner — APPROVED** [review]
Not much needed in terms of changes, just add some docs (preferably including a warning about CSS colors having mixed support depending on the terminal).

Just ignore the missing documentation lints on variants if you'd like, I plan on adding allow(missing_docs) to all the dynamic color variants.

**Comment by SirWindfield** [comment]
> As for "having colors in the root of the library" -> xterm and css colors don't go into the root of the library?

I missed this!

**Pull request merged**